controller/RestDelivery.php

```diff
@@ -212,6 +214,46 @@ public function updateDeferred()
         }
     }
 
+    /**
+     * Delete delivery by URI
+     */
+    public function deleteDeferred()
+    {
+        try {
+            if ($this->getRequestMethod() !== \Request::HTTP_DELETE) {
+                throw new \common_exception_NotImplemented('Only delete method is accepted to updating delivery');
+            }
+
+            if ($this->hasRequestParameter('uri')) {
```

Comment:
You throw the exception when the uri is presented. Please change it to throw the exception when the uri is not presented